internal/controller/multiclusterservice_controller.go

```diff
@@ -57,37 +67,71 @@ func (r *MultiClusterServiceReconciler) Reconcile(ctx context.Context, req ctrl.
 		return r.reconcileDelete(ctx, mcsvc)
 	}
 
-	if controllerutil.AddFinalizer(mcsvc, hmc.MultiClusterServiceFinalizer) {
-		if err := r.Client.Update(ctx, mcsvc); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to update MultiClusterService %s with finalizer %s: %w", mcsvc.Name, hmc.MultiClusterServiceFinalizer, err)
+	return r.reconcile(ctx, mcsvc)
+}
+
+func (r *MultiClusterServiceReconciler) reconcile(ctx context.Context, mcs *hmc.MultiClusterService) (res ctrl.Result, err error) {
+	var servicesErr error
```

Comment:
Yeah. To fix this, I would need to not use `err` in the return signature. So with the example you shared, it will become:
https://goplay.tools/snippet/qSblUFyHOu9

Reply:
The same as above applies to the reconciliation of ManagedCluster as well with regards to fetching the status of services.